src/Microsoft.AspNetCore.Razor.Language/DefaultRazorProjectEngine.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.AspNetCore.Razor.Language
+{
+    internal class DefaultRazorProjectEngine : RazorProjectEngine
+    {
+        public DefaultRazorProjectEngine(
+            RazorEngine engine,
+            RazorProjectFileSystem projectFileSystem,
+            IReadOnlyList<IRazorProjectEngineFeature> features)
+        {
+            if (engine == null)
+            {
+                throw new ArgumentNullException(nameof(engine));
+            }
+
+            if (projectFileSystem == null)
+            {
+                throw new ArgumentNullException(nameof(projectFileSystem));
+            }
+
+            if (features == null)
+            {
+                throw new ArgumentNullException(nameof(features));
+            }
+
+            Engine = engine;
+            ProjectFileSystem = projectFileSystem;
+            Features = features;
+
+            for (var i = 0; i < features.Count; i++)
+            {
+                features[i].ProjectEngine = this;
+            }
+        }
+
+        public override RazorProjectFileSystem ProjectFileSystem { get; }
```

Comment:
Can we call this `FileSystem` everywhere?

Reply:
Sure